core/consensus/babe/src/lib.rs

```diff
@@ -1043,6 +1120,16 @@ impl<B, E, Block, I, RA, PRA> BlockImport<Block> for BabeBlockImport<B, E, Block
 		// this way we can revert it if there's any error
 		let mut old_epoch_changes = None;
 
+		if let Some(enacted_epoch) = enacted_epoch.as_ref() {
```

Comment:
I was not sure - why cache has been updated only when both `next_epoch` and `enacted_epoch` are `Some()`. So reorganized code a bit